mindsdb/integrations/handlers/merlion_handler/merlion_handler.py

```diff
@@ -0,0 +1,193 @@
+    mses = MSESForecasterAdapter
+
+
+class TaskType(Enum):
+    detector = DetectorModelType
+    forecast = ForecastModelType
+
+
+def is_invalid_type(name: str, type_class: Enum) -> bool:
+    if name is None:
+        return True
+    return not name in type_class._member_names_
+
+
+def enum_to_str(type_class: Enum) -> str:
+    all = []
+    for element in type_class:
+        all.append(element.name)
+    return "|".join(all)
+
+
+def to_ts_dataframe(df: pd.DataFrame, time_col=None) -> str:
+    columns = list(df.columns.values)
+    if time_col is not None :
+        if time_col not in columns:
+            raise Exception("invalid column name: " + time_col)
+        if df[time_col].dtype != np.datetime64:
+            try:
+                df.index = pd.to_datetime(df[time_col])
+            except Exception as e:
+                raise Exception("can not convert column to datetime: " + time_col + " " + str(e))
+    else:
+        datetime_cols = list(df.select_dtypes(include=["datetime"]).columns.values)
+        if len(datetime_cols) > 0:
+            time_col = datetime_cols[0]
+            df.index = pd.to_datetime(df[time_col])
+        if time_col is None:
+            raise Exception("can not find datetime column for time series")
+    df.drop(columns=[time_col], inplace=True)
+    return time_col
+
+
+class MerlionHandler(BaseMLEngine):
+    name = 'merlion'
+
+    ARG_USING_TASK = "task"
+    ARG_USING_MODEL_TYPE = "model_type"
+    ARG_USING_TIME_COLUMN = "time_column"
+    ARG_WINDOW = "window"
+    ARG_TARGET = "target" # only be used to persist args to args.json
+    ARG_COLUMN_SEQUENCE = "column_sequence"  # only be used to persist args to args.json
+
+    KWARGS_DF = "df"
+
+    DEFAULT_MODEL_TYPE = "default"
+    DEFAULT_MAX_PREDICT_STEP = 100
+    DEFAULT_PREDICT_BASE_WINDOW = 10
+
+    PERSISIT_MODEL_FILE_NAME = "merlion_model"
+    PERSISIT_ARGS_KEY_IN_JSON_STORAGE = "args"
+
+    def create(self, target, args=None, **kwargs):
+        df: pd.DataFrame = kwargs.get(self.KWARGS_DF, None)
+        # prepare arguments
+        task = args.get(self.ARG_USING_TASK, TaskType.forecast.name)
+        model_type = args.get(self.ARG_USING_MODEL_TYPE, self.DEFAULT_MODEL_TYPE)
+        time_column = args.get(self.ARG_USING_TIME_COLUMN, None)
+        horizon = args.get("horizon", self.DEFAULT_MAX_PREDICT_STEP)
+        window = args.get(self.ARG_WINDOW, self.DEFAULT_PREDICT_BASE_WINDOW)
+        # update args for default value maybe has been used, only time column will be set afterwards
+        args[self.ARG_TARGET] = target
+        args[self.ARG_USING_TASK] = task
+        args[self.ARG_USING_MODEL_TYPE] = model_type
+        args["horizon"] = horizon
+        args[self.ARG_WINDOW] = window
+
+        # check df
+        if df is None:
+            raise Exception("missing required key in args: " + self.KWARGS_DF)
+        else:
+            column_sequence = list(df.columns.values)
+            sorted(column_sequence)
```

Comment:
This isn't doing anything. The in-place variant is `list.sort()`. However, as mentioned above, better to avoid side effects, so I'd ask for:

```
column_sequence = sorted(list(df.columns.values))
```